src/Compilers/Core/Portable/Syntax/SyntaxDiffer.cs

```diff
@@ -500,11 +496,43 @@ private static bool AreIdentical(in SyntaxNodeOrToken node1, in SyntaxNodeOrToke
             return node1.UnderlyingNode == node2.UnderlyingNode;
         }
 
-        private static bool AreSimilar(in SyntaxNodeOrToken node1, in SyntaxNodeOrToken node2)
+        private bool AreSimilar(in SyntaxNodeOrToken node1, in SyntaxNodeOrToken node2)
         {
-            return node1.RawKind == node2.RawKind;
+            if (node1.RawKind == node2.RawKind)
+            {
+                return true;
+            }
+
+            if (node1.IsToken || node2.IsToken)
+            {
+                return false;
+            }
+
+            return AreSimilar(node1.AsNode(), node2.AsNode());
         }
 
+        private bool AreSimilar(SyntaxNode node1, SyntaxNode node2)
```

Comment:
>AreSimilar [](start = 21, length = 10)

consider making this a local function and removing redundant check for `RawKind`